OvernightPrintsSpecBindings/Utils/Utils.cs

```diff
@@ -22,38 +22,53 @@ public static string ParseString(string input)
 			return input.Trim('"');
 		}
 
-		public static string Resolve(String email)
-		{
-			string result = String.Empty;
-			string regex = "(?<={{)(.*?)(?=}})";
-			string match = Regex.Match(email, regex).ToString();
+		private static string vocabulary = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!$%^&?*-+={}";
+
+		private static string regexAll = "(?={{)(.*?)(?<=}})";
+		private static string regexLeft = "(?<={{)(.*?)(?=::)";
+		private static string regexRight = "(?<=::)(.*?)(?=}})";
+
+		private delegate string Function(string email);
 
-			string key = match.Split("::".ToCharArray())[0];
-			string value = match.Split("::".ToCharArray())[2];
+		private static Function[] functions = { new Function(rnd), new Function(context)};
 
-			switch (key)
+		public static string Resolve(string email)

```

Comment:
Use this method to obtain input string, get placeholder, handle it and combine input and result of rnd and context. Try to replace the whole "{{a::b}}" placeholder with result of rnd or context methods